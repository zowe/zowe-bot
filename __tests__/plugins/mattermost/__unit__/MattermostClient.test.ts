/*
 * This program and the accompanying materials are made available under the terms of the
 * Eclipse Public License v2.0 which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Copyright Contributors to the Zowe Project.
 */

import { SuperAgentRequest } from 'superagent';
import { MattermostClient } from '../../../../src/plugins/mattermost/MattermostClient';
import { MattermostMiddleware } from '../../../../src/plugins/mattermost/MattermostMiddleware';
import { IMattermostOption } from '../../../../src/types';

describe('MMClient unit', () => {
  it('empty, blank, or null tlsOptions', () => {
    const snapAgent = (agent: SuperAgentRequest, testName: string) => {
      // agent._ca tracks the loaded cert authority, and is not dumped as part of normal object
      expect((agent as any)._ca).toMatchSnapshot(testName);
    };

    const tests: { name: string; ca: any; validate: (agent: SuperAgentRequest, testName: string) => void }[] = [
      { name: 'ca 0spaces', ca: '', validate: snapAgent },
      { name: 'ca 3spaces', ca: '  ', validate: snapAgent },
      { name: 'ca manySpaces', ca: '          ', validate: snapAgent },
      { name: 'ca null', ca: null, validate: snapAgent },
      { name: 'ca undefined', ca: undefined, validate: snapAgent },
      { name: 'realPath', ca: '/bad/path', validate: snapAgent },
    ];

    for (const test of tests) {
      const client = new MattermostClient(
        {} as MattermostMiddleware,
        {
          protocol: 'https',
          tlsCertificate: test.ca,
        } as IMattermostOption,
      );

      const postReq = client.post('badurl');
      test.validate(postReq, 'post_' + test.name);
      const getReq = (client as any).buildGet('badurl');
      test.validate(getReq, 'get_' + test.name);
    }
  });
});
