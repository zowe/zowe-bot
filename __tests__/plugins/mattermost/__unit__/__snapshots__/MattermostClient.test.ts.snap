// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MM client unit test empty, blank, or null tlsOptions: get_ca 0spaces 1`] = `
"{"method":"GET","url":"badurl","headers":{"authorization":"BEARER undefined","accept":"application/json","content-type":"application/json"}}
undefined"
`;

exports[`MM client unit test empty, blank, or null tlsOptions: get_ca 3spaces 1`] = `
"{"method":"GET","url":"badurl","headers":{"authorization":"BEARER undefined","accept":"application/json","content-type":"application/json"}}
undefined"
`;

exports[`MM client unit test empty, blank, or null tlsOptions: get_ca manySpaces 1`] = `
"{"method":"GET","url":"badurl","headers":{"authorization":"BEARER undefined","accept":"application/json","content-type":"application/json"}}
undefined"
`;

exports[`MM client unit test empty, blank, or null tlsOptions: get_ca null 1`] = `
"{"method":"GET","url":"badurl","headers":{"authorization":"BEARER undefined","accept":"application/json","content-type":"application/json"}}
undefined"
`;

exports[`MM client unit test empty, blank, or null tlsOptions: get_ca undefined 1`] = `
"{"method":"GET","url":"badurl","headers":{"authorization":"BEARER undefined","accept":"application/json","content-type":"application/json"}}
undefined"
`;

exports[`MM client unit test empty, blank, or null tlsOptions: get_realPath 1`] = `
"{"method":"GET","url":"badurl","headers":{"authorization":"BEARER undefined","accept":"application/json","content-type":"application/json"}}
/bad/path"
`;

exports[`MM client unit test empty, blank, or null tlsOptions: post_ca 0spaces 1`] = `
"{"method":"POST","url":"badurl","headers":{"authorization":"BEARER undefined","accept":"application/json","content-type":"application/json"}}
undefined"
`;

exports[`MM client unit test empty, blank, or null tlsOptions: post_ca 3spaces 1`] = `
"{"method":"POST","url":"badurl","headers":{"authorization":"BEARER undefined","accept":"application/json","content-type":"application/json"}}
undefined"
`;

exports[`MM client unit test empty, blank, or null tlsOptions: post_ca manySpaces 1`] = `
"{"method":"POST","url":"badurl","headers":{"authorization":"BEARER undefined","accept":"application/json","content-type":"application/json"}}
undefined"
`;

exports[`MM client unit test empty, blank, or null tlsOptions: post_ca null 1`] = `
"{"method":"POST","url":"badurl","headers":{"authorization":"BEARER undefined","accept":"application/json","content-type":"application/json"}}
undefined"
`;

exports[`MM client unit test empty, blank, or null tlsOptions: post_ca undefined 1`] = `
"{"method":"POST","url":"badurl","headers":{"authorization":"BEARER undefined","accept":"application/json","content-type":"application/json"}}
undefined"
`;

exports[`MM client unit test empty, blank, or null tlsOptions: post_realPath 1`] = `
"{"method":"POST","url":"badurl","headers":{"authorization":"BEARER undefined","accept":"application/json","content-type":"application/json"}}
/bad/path"
`;
